Database set up

cshih299 opened the pull request.

cshih299 committed:
functionality to create the database and pull in twitter statistics

cshih299 committed:
Merge branch 'master' of https://github.com/strburst/twitter-streaming-stats into db

strburst commented on example.js:
```diff
@@ -1,9 +1,11 @@
 #!/usr/bin/env node
 /* eslint no-console: off */
 
+
```
Remove me!

strburst commented on example.js:
```diff
@@ -19,6 +21,13 @@ class CheckFeatures {
+        console.log('hashtag: ' + hashtagList[i]);
+        db.insertHashtagStats(tweet.place.country, hashtagList[i].text);
+      }
+    }
```
You should create your own processor object if you're adding new stuff, instead of modifying `CheckFeatures`. I was intending `CheckFeatures` to get some stats on what information is included in tweets.

strburst commented on db.js:
```diff
@@ -1,16 +1,130 @@
+    type: Sequelize.INTEGER,
+    defaultValue: 1
+  },
+});
```
I think each processor should have its own table that it knows about—I would put the stuff defining table structure for each metric into the code for the processor. E.g. this would go with the new `HashtagStats` processor you're going to create.

strburst commented on example.js:
```diff
@@ -19,6 +21,13 @@ class CheckFeatures {
+    }
     if (tweet.geo) {
       this.placeCount += 1;
     }
```
It looks like we're making this file the starting point for the batch scanning job, instead of just an example. Maybe rename this file `scan.js` or `scanjob.js` or something.

strburst commented on db.js:
```diff
@@ -1,16 +1,130 @@
+  },
+});
+
+// returns a list of all the trending hashtags and counties
```
Typo ("counties"). Also, maybe use the Javadoc `/** */` convention I'm using in `util.js` for all the function-level comments.

strburst commented on db.js:
```diff
@@ -1,16 +1,130 @@
+//   .command('drop', 'Deletes the database', {}, drop)
+//   .demandCommand(1, 1, 'Must provide exactly one subcommand')
+//   .help()
+//   .argv;
```
Uncomment this before we merge.

strburst commented on db.js:
```diff
@@ -1,16 +1,130 @@
+        });
+      }
+    });
+}
```
This would work, but I would think about making this so that it only writes to the database every 1000 tweets or so.

strburst commented on db.js:
```diff
@@ -1,16 +1,130 @@
+
+}
+
+// inserts and updates hashtag statistics in HashtagStats
```
Same here.

strburst reviewed (changes requested):
Make sure you run the linter and clean up all the style errors before you're done. I'd be willing to start working on this soon even if it's not perfect; just let me know when you're mostly satisfied in the code. (I renamed this to have [WIP] in the title.)

strburst commented on db.js:
```diff
@@ -1,16 +1,130 @@
 /* eslint no-unused-expressions: off */
 
-function create() {
+// set up the connection with db
```
I usually start comments with an uppercase letter, but that's just my preference.

strburst committed:
Merge branch 'master' into db

strburst committed:
Fix various lint and style issues

strburst committed:
Don't do argument parsing when required

strburst committed:
Refactor config to also handle db configs

Custom configuration file is now 'custom.js(on)?', and should look have
'db' and 'keys' sections.

strburst committed:
Make sure we really exit after SIGINT

strburst committed:
Rename example.js to scanjob.js

example.js is no longer just an example, but is where we're putting the
background process that scans tweets.

strburst committed:
Fix wrong config section in db.js

strburst committed:
Reorganize tweetprocessors and db code

Make each processor responsible for knowing the schema it wants to write
to. Put all processor classes in their own file.

strburst committed:
Use country code instead of country name

This will make things easier to index.
